File: Desktop/application/frontend/models/Predmet.ts
import { Klijent } from "./Klijent";
import { Parcela } from "./Parcela";
import { Posao } from "./Posao";

export class Predmet {
    id: number;
    parcela: Parcela;
    klijent: Klijent;
    posao: Posao;
    datum: Date;
    cena: number;
    dug: number;
    zavrsen: boolean;

    constructor() {
        this.id = 0;
        this.posao = { id: 0, naziv: '' };
        this.klijent = {
            id: 18,
            ime: '',
            imeOca: '',
            prezime: '',
            jmbg: '',
            brlk: '',
            pribivaliste: '',
            telefon: ''
        };
        this.parcela = {
            id: 0,
            brKatParcele: '',
            katOpstina: '',
            opstina: '',
            ln: null
        };
        this.cena = null;
        this.datum = null;
        this.zavrsen = false;
        this.dug = 0;
    }
}

File: Desktop/application/backend/entity/Predmet.java
package com.example.demo.entity;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;
import java.util.Optional;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder

public class Predmet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;



    @ManyToOne
    @JoinColumn(
            name = "idparcela",
            referencedColumnName = "id"
    )
    private Parcela parcela;

    @ManyToOne
    @JoinColumn(
            name = "idklijent",
            referencedColumnName = "id"
    )
    private Klijent klijent;


    @ManyToOne
    @JoinColumn(
            name = "idposao",
            referencedColumnName = "id"
    )
    private Posao posao;


    @Column
    private Date datum;

    @Column(nullable = false)
    private double cena;

    @Column(nullable = false)
    private double dug;


    @Column
    private boolean zavrsen;




}
